[Phoon Jun Hui] iP

junhui-phoon opened the pull request.

Commit:
Add Gradle support

Commit:
added Duke Level-1 functionalities

Commit:
added Duke Level-2 functionalities

Commit:
added Duke Level-3 functionalities

Commit:
added Duke Level-4 functionalities

Commit:
Update runtest.sh; EXPECTED.TXT; input.txt

Commit:
added Duke Level-5 functionalities

Commit:
added Duke Level-6 functionalities

Commit:
update input.txt, EXPECTED.TXT; fix typo in Duke.java

Commit:
Duke.java & Task.java: level-7 in-progress

Commit:
Add Duke Level-8 functionalities

Commit:
Add Duke Level-7 functionalities

Commit:
Merge branch 'master' into branch-Level-8

# Conflicts:
#	src/main/java/Deadline.java
#	src/main/java/Duke.java
#	src/main/java/Event.java
#	src/main/java/Task.java
#	src/main/java/Todo.java

Commit:
A-MoreOOP in progress

Commit:
A-MoreOOP in progress

Commit:
A-MoreOOP in progress

Commit:
Add Duke A-MoreOOP increment

Commit:
Add A-Packages increment

Commit:
runtest.sh : update javac classpath

Commit:
Add JUnit Tests

Commit:
Add Javadoc to all classes.

Commit:
Update all classes to follow CodingStandard

Commit:
Update all classes with proper indentation

Commit:
Update all classes with proper indentation

Commit:
Update all classes to proper indentation

Commit:
Update all classes to adhere coding standard

Commit:
Add Duke Level-9 increment

Commit:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/Todo.java
#	src/main/java/duke/ui/Ui.java

Commit:
Merge branch 'master' into branch-A-CodingStandard

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/Todo.java
#	src/main/java/duke/ui/Ui.java

Commit:
Merge branch 'master' into branch-Level-9

# Conflicts:
#	src/main/java/duke/Duke.java
#	src/main/java/duke/parser/Parser.java
#	src/main/java/duke/storage/Storage.java
#	src/main/java/duke/task/Deadline.java
#	src/main/java/duke/task/Event.java
#	src/main/java/duke/task/Task.java
#	src/main/java/duke/task/TaskList.java
#	src/main/java/duke/task/Todo.java
#	src/main/java/duke/ui/Ui.java

Commit:
Merge branch 'branch-Level-9'

Commit:
Ui.java : add a line in showHelp()

Commit:
runtest.sh : Update classpath

Commit:
Update text-ui-text to include up to Level-9 tests

Nauw1010 commented on src/main/java/duke/Duke.java:
```diff
@@ -45,7 +45,7 @@ public Duke(String filePath) {
         String userInput;
         String[] userInputArray;
-        boolean validInput = true;
+        boolean validInput;
```
Perhaps the boolean should be "isValidInput" instead to sounds like a boolean I think?

Commit:
Merge branch 'master' into add-gradle-support

Commit:
Merge branch 'add-gradle-support' into branch-A-Gradle

Commit:
Add Gradle support to project

Commit:
Add Gradle files

Commit:
Merge branch 'master' into branch-A-Gradle

# Conflicts:
#	gradle/wrapper/gradle-wrapper.jar
#	gradle/wrapper/gradle-wrapper.properties
#	gradlew
#	gradlew.bat

Commit:
Merge branch 'branch-A-Gradle'

Commit:
Level-10 GUI implementation in progress

Commit:
Add runGui() in Duke.java for GUI support (In progress)

Nauw1010 reviewed (commented):
Nice work!

Commit:
Update Level-10 GUI implementation (In progress)

Commit:
Add Level-10 GUI implementation

Commit:
Merge branch 'branch-Level-10'

LeeMingDe commented on src/main/java/duke/Launcher.java:
```diff
@@ -0,0 +1,11 @@
+
+import javafx.application.Application;
+
+public class Launcher {
```
Think you can add java docs header about what this class does. Eg, Responsible for launching javafx interface

LeeMingDe commented on src/main/java/duke/parser/Parser.java:
```diff
@@ -0,0 +1,71 @@
+        return taskNumbers;
+    }
+
+    public static String[] parseCommand(String inputString) {
```
I think you can add in java docs for this method!

LeeMingDe commented on src/main/java/duke/task/TaskList.java:
```diff
@@ -0,0 +1,98 @@
+        return this.tasks;
+    }
+
+    public TaskList findTasks(String keyword) throws DukeException {
```
Can add in java docs here too!

LeeMingDe commented on src/main/java/duke/ui/Gui.java:
```diff
@@ -0,0 +1,30 @@
+import javafx.scene.layout.AnchorPane;
+import javafx.stage.Stage;
+
+public class Gui extends Application {
```
Same here for java docs!

LeeMingDe commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,485 @@
+
+import duke.parser.Parser;
+
+public class Duke {
```
Can add java docs here too!

LeeMingDe commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,485 @@
+
+public class Duke {
+    public static String home = System.getProperty("user.home");
+    public static String filePath = home + "/Desktop/duke.txt";
```
i think it would be helpful to leave a line  to separate the static variables with the non-static ones.

LeeMingDe commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,485 @@
+                break;
+            }
+            switch (userInput) {
+            case "list":
```
i think you can abstract all the methods out because your line of code is slightly long? >30 lines

LeeMingDe commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,485 @@
+        String[] GuiInputArray;
+        String[] userInputArray;
+
+        userInput = ui.readInput(isGui);
```
Again, i think you can abstract all the methods out so that your code is easier to read!

LeeMingDe commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,485 @@
+                    ;
+                    do {
+                        ui.printMsg(ui.showListDoneAskMsg());
+                        try {
```
there is alot of statements nested inside your code so it somewhat has an arrowhead format, will be good to abstract the methods out!

LeeMingDe reviewed (commented):
Overall, i think your code is good, with most of the methods and classes having headers and java docs. However, it would have been better if you could abstract some of the code out for better readability.

Commit:
Fix DialogBox height

Commit:
Add java assert to Parser.java

Commit:
Fix parse() method to parseMultipleTasks() method for clarity

Java streams was implemented before week 5

Commit:
Add javadoc to parseCommand() method

junhui-phoon committed:
Merge pull request #2 from junhui-phoon/branch-A-Streams

Add A-Streams to Duke

Commit:
Merge branch 'master' into branch-A-Assertions

Commit:
Refactor runGui() method in Duke.java for readability

Each switch case is refactored into different methods.

Commit:
Merge branch 'master' into branch-A-CodeQuality

# Conflicts:
#	src/main/java/duke/Duke.java

Commit:
Improve Code Quality

Fixed typo 'validInput' to 'isValidInput'.
Removed redundant code.
Added missing javadoc for several methods.

Commit:
Remove Duke CLI functionality

run() method in Duke.java is removed.

junhui-phoon committed:
Merge pull request #3 from junhui-phoon/branch-A-CodeQuality

Add A-CodeQuality to Duke

Commit:
Merge branch 'master' into branch-A-Assertions

Commit:
Add more java assert statements in Duke.java

junhui-phoon committed:
Merge pull request #1 from junhui-phoon/branch-A-Assertions

Add java assert to Parser.java, Duke.java

Commit:
Merge branch 'master' into branch-B-ViewSchedules

Commit:
Add B-ViewSchedules functionality

Added B-ViewSchedules functionality to Duke using 'date' command.
Removed redundant methods in UI.java due to removal of Duke CLI interface.

junhui-phoon committed:
Merge pull request #4 from junhui-phoon/branch-B-ViewSchedules

Add B-ViewSchedules functionality

Commit:
Improve the GUI

junhui-phoon committed:
Set theme jekyll-theme-midnight

Commit:
Fix JavaFX version in DialogBox.fxml

Commit:
Add list_outcome.png for docs/README.md

Commit:
Add a User Guide

Fix typo in Deadline.java, Event.java

Commit:
Fix UTF-8 characters

Commit:
Remove UTF-8 characters in README.md
